libs/providers/ofrep-web/src/lib/ofrep-web-provider.ts

```diff
@@ -260,37 +269,41 @@ export class OFREPWebProvider implements Provider {
   }
 
   /**
-   * Evaluate is a function retrieving the value from a flag in the cache.
+   * _resolve is a function retrieving the value from a flag in the cache.
    * @param flagKey - name of the flag to retrieve
    * @param type - type of the flag
+   * @param defaultValue - default value
    * @private
    */
-  private evaluate<T extends FlagValue>(flagKey: string, type: string): ResolutionDetails<T> {
+  private _resolve<T extends FlagValue>(flagKey: string, type: string, defaultValue: T): ResolutionDetails<T> {
     const resolved = this._flagCache[flagKey];
+
     if (!resolved) {
-      throw new FlagNotFoundError(`flag key ${flagKey} not found in cache`);
+      return {
+        value: defaultValue,
+        flagMetadata: this._flagSetMetadataCache,
+        reason: StandardResolutionReasons.ERROR,
+        errorCode: ErrorCode.FLAG_NOT_FOUND,
+        errorMessage: ErrorMessageMap[ErrorCode.FLAG_NOT_FOUND],
+      };
     }
 
     if (isResolutionError(resolved)) {
-      switch (resolved.errorCode) {
-        case EvaluationFailureErrorCode.FlagNotFound:
-          throw new FlagNotFoundError(`flag key ${flagKey} not found: ${resolved.errorDetails}`);
-        case EvaluationFailureErrorCode.TargetingKeyMissing:
-          throw new TargetingKeyMissingError(`targeting key missing for flag key ${flagKey}: ${resolved.errorDetails}`);
-        case EvaluationFailureErrorCode.InvalidContext:
-          throw new InvalidContextError(`invalid context for flag key ${flagKey}: ${resolved.errorDetails}`);
-        case EvaluationFailureErrorCode.ParseError:
-          throw new ParseError(`parse error for flag key ${flagKey}: ${resolved.errorDetails}`);
-        case EvaluationFailureErrorCode.General:
-        default:
-          throw new GeneralError(
-            `general error during flag evaluation for flag key ${flagKey}: ${resolved.errorDetails}`,
-          );
-      }
+      return {
+        ...resolved,
+        value: defaultValue,
+        errorMessage: ErrorMessageMap[resolved.errorCode],
+      };
     }
 
     if (typeof resolved.value !== type) {
-      throw new TypeMismatchError(`flag key ${flagKey} is not of type ${type}`);
+      return {
+        value: defaultValue,
+        flagMetadata: this._flagSetMetadataCache,
```

Comment:
Please use flag metadata not flag set metadata.

Reply:
Good call.